packages/beeq/src/components/date-picker/bq-date-picker.tsx

```diff
@@ -0,0 +1,537 @@
+    const { value } = ev.target as unknown as { value: string };
+
+    this.value = value;
+    this.formattedDate = this.formatDate(this.value);
+    this.inputElem.value = this.formattedDate;
+
+    this.bqChange.emit({ value: this.value, el: this.el });
+
+    this.open = this.type === 'multi';
+  };
+
+  private handleClearClick = (ev: CustomEvent) => {
+    if (this.disabled) return;
+
+    this.inputElem.value = '';
+    this.value = this.inputElem.value;
+
+    this.bqClear.emit(this.el);
+    this.bqChange.emit({ value: this.value, el: this.el });
+    this.inputElem.focus();
+
+    ev.stopPropagation();
+  };
+
+  private handleLabelSlotChange = () => {
+    this.hasLabel = hasSlotContent(this.labelElem);
+  };
+
+  private handlePrefixSlotChange = () => {
+    this.hasPrefix = hasSlotContent(this.prefixElem);
+  };
+
+  private handleSuffixSlotChange = () => {
+    this.hasSuffix = hasSlotContent(this.suffixElem);
+  };
+
+  private generateCalendarMonth = (offset?: number, className = ''): JSX.Element => {
+    return (
+      <calendar-month
+        offset={offset}
+        class={className}
+        exportparts={`${this.COMMON_EXPORT_PARTS},${this.BUTTON_EXPORT_PARTS}`}
+      />
+    );
+  };
+
+  /**
+   * Generates an array of JSX.Element representing calendar months.
+   *
+   * If the type of the date picker is 'range' or 'multi' and the number of months is specified,
+   * it generates an array of calendar months with the specified length. Each month will have an offset
+   * and a class name based on its position in the array. The offset is used to determine the month to display,
+   * and the class name is used for responsive design.
+   *
+   * If the type of the date picker is not 'range' or 'multi', or if the number of months is not specified,
+   * it generates an array with a single calendar month.
+   *
+   * @returns {JSX.Element[]} An array of JSX.Element representing calendar months.
+   */
+  private generateCalendarMonths = (): JSX.Element[] => {
+    if (this.type === 'range' || (this.type === 'multi' && this.months)) {
+      return Array.from({ length: this.months }, (_, i) => {
+        const offset = i > 0 ? i : undefined;
+        const className = offset ? 'hidden sm:block' : '';
+        return this.generateCalendarMonth(offset, className);
+      });
+    }
+
+    return [this.generateCalendarMonth()];
+  };
+
+  /**
+   * Extracts and returns the last date part from a given string.
+   * When the type of the date picker is 'range' or 'multi', the last date part of the value
+   * should be the focused date in the calendar.
+   *
+   * @param value - The value to be processed, can be a string.
+   * @returns The extracted last date portion of the value.
+   */
+  private focusedDate = (value: string) => {
+    if (!value) return;
+
+    const dateLength = 10; // Length of a standard date in the format YYYY-MM-DD
+    return value.slice(-dateLength);
+  };
```

Comment:
We need to set the `focusedDate` to the first instance of `yyyy-mm-dd` of the value string, otherwise in a range or multi variant the user can not see where the initial value of the range is set:

![CleanShot 2024-05-20 at 18 47 39@2x](https://github.com/Endava/BEEQ/assets/328492/9ba55dcb-08a3-4c55-b520-97bb75004e4c)

```suggestion
  /**
   * Extracts and returns the first date part from a given string.
   * When the type of the date picker is 'range' or 'multi', the first or initial date part of the value
   * should be the focused date in the calendar.
   *
   * @param value - The value to be processed, can be a string.
   * @returns The extracted last date portion of the value.
   */
  private focusedDate = (value: string) => {
    if (!value) return;

    const dateRegex = /\b\d{4}-\d{2}-\d{2}\b/;
    const match = dateRegex.exec(value);
    return match ? match[0] : null;
  };
```

![CleanShot 2024-05-20 at 18 51 28@2x](https://github.com/Endava/BEEQ/assets/328492/0c85aa06-2e8e-41d7-956c-7cae33b98120)